src/posit/connect/client.py

```diff
@@ -192,10 +197,50 @@ def with_user_session_token(self, token: str) -> Client:
         Client
             A new Client instance authenticated with an API key exchanged for the user session token.
 
+        Raises
+        ------
+        ValueError
+            If the provided token is `None` or empty or if the exchange response is malformed.
+        ClientError
+            If the token exchange request with the Connect Server fails.
+
         Examples
         --------
         >>> from posit.connect import Client
         >>> client = Client().with_user_session_token("my-user-session-token")
+
+        Example using user session token from Shiny session:
+        >>> from posit.connect import Client
+        >>> from shiny.express import render, session
+        >>>
+        >>> client = Client()
+        >>>
+        >>> @reactive.calc
+        >>> def visitor_client():
+        ...     ## read the user session token and generate a new client
+        ...     user_session_token = session.http_conn.headers.get(
+        ...         "Posit-Connect-User-Session-Token"
+        ...     )
+        ...     return client.with_user_session_token(user_session_token)
+        >>> @render.text
+        >>> def user_profile():
+        ...     # fetch the viewer's profile information
+        ...     return visitor_client().me
+
+        Example with fallback when header is missing in API request:
+        >>> from posit.connect import Client
+        >>> import requests
+        >>>
+        >>> def get_client(request):
+        ...     base_client = Client()
```

Comment:
What credentials would this `base_client` be authenticated with?

Reply:
CONNECT_URL and CONNECT_API_KEY. I don't think it is necessary to restate that here since it's mentioned in the Client documentation.

Reply:
Ah, good point. Yes, it's better to raise the exception here instead of returning the original client.